Fails when submitting a listed add-on

Version: 0.2.2
Runs on: ubuntu-latest on GitHub Actions

The publish step fails when I submit [my add-on](https://github.com/iorate/uBlacklist) to AMO. Although a new version is actually uploaded for review, the publish step fails, so the subsequent plugins are not executed.

`web-ext` seems to throw an error when submitting a listed add-on (ref. https://github.com/mozilla/web-ext/issues/804 ).

Could you please work around it? (like https://github.com/fregante/web-ext-submit ?)

Log:
```
[3:08:17 PM] [semantic-release] › ℹ  Start step "publish" of plugin "semantic-release-firefox-add-on"
Building web extension from dist/firefox/production

Validating add-on [            .                                               ]
...
Validating add-on [............................................................]
Validation results: https://addons.mozilla.org/en-US/developers/upload/********************************

Signing add-on [           .                                                   ]
Signing add-on [...............................................................]
Your add-on has been submitted for review. It passed validation but could not be automatically signed because this is a listed add-on.
FAIL
[3:08:23 PM] [semantic-release] › ✖  Failed step "publish" of plugin "semantic-release-firefox-add-on"
[3:08:23 PM] [semantic-release] › ✖  An error occurred while running semantic-release: o: The extension could not be signed
    at /home/runner/work/uBlacklist/uBlacklist/node_modules/web-ext/dist/web-ext.js:1:66345
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at publish (/home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release-firefox-add-on/src/publish.js:25:42)
    at validator (/home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release/lib/plugins/normalize.js:34:24)
    at /home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release/lib/plugins/pipeline.js:37:34
    at async Promise.all (index 0)
    at next (/home/runner/work/uBlacklist/uBlacklist/node_modules/p-reduce/index.js:16:18) {
  pluginName: 'semantic-release-firefox-add-on'
}
o: The extension could not be signed
    at /home/runner/work/uBlacklist/uBlacklist/node_modules/web-ext/dist/web-ext.js:1:66345
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at publish (/home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release-firefox-add-on/src/publish.js:25:42)
    at validator (/home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release/lib/plugins/normalize.js:34:24)
    at /home/runner/work/uBlacklist/uBlacklist/node_modules/semantic-release/lib/plugins/pipeline.js:37:34
    at async Promise.all (index 0)
    at next (/home/runner/work/uBlacklist/uBlacklist/node_modules/p-reduce/index.js:16:18) {
  pluginName: 'semantic-release-firefox-add-on'
}
##[error]Process completed with exit code 1.
```

--- a/README.md
+++ b/README.md
@@ -73,7 +73,7 @@ This plugin requires some parameters to be set, so be sure to check below and fi
 
 ### `publish`
 
-Creates an unsigned XPI file out of the source directory and uploads it to the Mozilla Add On, using the web-ext sign command. The output from the sign command will be passed through to the console. If the package is validated and signed, it will downloaded the signed XPI file and store it in the artifacts directory under the specified file name.
+Creates an unsigned XPI file out of the source directory and uploads it to the Mozilla Add On, using the web-ext sign command. The output from the sign command will be passed through to the console. If the package is validated and signed, it will download the signed XPI file and store it in the artifacts directory under the specified file name. If the package is validated but not signed (including the case where manual review is required), it will store the unsigned XPI file in the artifacts directory.
 
 #### `publish` parameters
 

--- a/src/publish.js
+++ b/src/publish.js
@@ -2,6 +2,7 @@ const fs = require('fs')
 const path = require('path')
 
 const webExt = require('web-ext').default
+const defaultAddonSigner = require('sign-addon')
 
 const { verifyOptions } = require('./utils')
 
@@ -22,14 +23,29 @@ const publish = async options => {
     } = verifyOptions(options, ['extensionId', 'targetXpi'])
 
     const { FIREFOX_API_KEY, FIREFOX_SECRET_KEY } = process.env
-    const { success, downloadedFiles } = await webExt.cmd.sign({
-        apiKey: FIREFOX_API_KEY,
-        apiSecret: FIREFOX_SECRET_KEY,
-        artifactsDir,
-        channel,
-        id: extensionId,
-        sourceDir,
-    })
+
+    let unsignedXpiPath
+    const signAddon = async params => {
+        unsignedXpiPath = params.xpiPath
+        const result = await defaultAddonSigner(params)
+        if (!result.success && result.errorCode === 'ADDON_NOT_AUTO_SIGNED') {
+            result.success = true
+            result.downloadedFiles = result.downloadedFiles || []
+        }
+        return result
+    }
+
+    const { success, downloadedFiles } = await webExt.cmd.sign(
+        {
+            apiKey: FIREFOX_API_KEY,
+            apiSecret: FIREFOX_SECRET_KEY,
+            artifactsDir,
+            channel,
+            id: extensionId,
+            sourceDir,
+        },
+        { signAddon },
+    )
     if (!success) {
         throw new Error(
             'Signing the extension failed. See the console output from web-ext sign for the validation link',
@@ -37,7 +53,7 @@ const publish = async options => {
     }
     const [xpiFile] = downloadedFiles
     fs.renameSync(
-        path.join(artifactsDir, xpiFile),
+        xpiFile ? path.join(artifactsDir, xpiFile) : unsignedXpiPath,
         path.join(artifactsDir, targetXpi),
     )
 }

